### PR TITLE
Test Create Primitive Controller Input

### LINKED ISSUE #1744: Test ControllerPrimitiveGenerator::createPrimitiveFromControllerInput
### Description of the task
Add unit tests to `createPrimitiveFromControllerInput` in the `ControllerPrimitiveGenerator` class (controller_primitive_generator.cpp)

### Acceptance criteria

- [ ] Add unit tests
- [ ] Make sure it works

### PR DESCRIPTION
**Pull request by BrianS99**
### Description
Wrote test cases to test `createPrimitiveFromControllerInput()` and ensuring they return the correct primitives for a specific button pressed on the handheld controller. Made the function public and static in `controller_input_generator.h` to make it actually testable. 

### Testing Done
Added 3 test cases to `controller_primitive_generator_test.cpp`. All pass

### Resolved Issues
Resolves #1744

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/handheld_controller/controller_primitive_generator_test.cpp [hunk 1]
```diff
@@ -28,3 +31,67 @@ TEST(ControllerPrimitiveGeneratorTest, test_create_direct_velocity)
               0.5);
     EXPECT_EQ(direct_velocity_primitive->direct_control().dribbler_speed_rpm(), 200);
 }
+
+TEST(ControllerPrimitiveGeneratorTest, test_create_primitive_controller_input)
+{
+    ControllerInput input = ControllerInput();
```
Comment (jonathanlew) [inline src/software/handheld_controller/controller_primitive_generator_test.cpp]: can we add a brief comment at the top of every test to explain what's happening?

File: src/software/handheld_controller/controller_primitive_generator_test.cpp [hunk 2]
```diff
@@ -28,3 +31,67 @@ TEST(ControllerPrimitiveGeneratorTest, test_create_direct_velocity)
...
+        DynamicParameters->getHandheldControllerConfig()->ChipDistanceMeters()->value();
+    auto expected_chip_primitive =
+        *createChipPrimitive(Point(0, 0), Angle::zero(), chip_distance);
+    EXPECT_TRUE(google::protobuf::util::MessageDifferencer::Equals(
+        expected_chip_primitive, actual_primitive));
+}
+
+TEST(ControllerPrimitiveGeneratorTest, test_create_primitive_controller_input2)
+{
+    ControllerInput input = ControllerInput();
```
Comment (jonathanlew) [inline src/software/handheld_controller/controller_primitive_generator_test.cpp:83]: can we actually set x/y/angular velocities for this input?

**Review by jonathanlew — CHANGES_REQUESTED** [review]

File: src/software/handheld_controller/controller_primitive_generator_test.cpp (hunk 1 above)
Comment (BrianS99) [reply src/software/handheld_controller/controller_primitive_generator_test.cpp]: Ok, will do

File: src/software/handheld_controller/controller_primitive_generator_test.cpp (hunk 2 above)
Comment (BrianS99) [reply src/software/handheld_controller/controller_primitive_generator_test.cpp:83]: I'm not sure, I haven't tried that but I think the function `controller_primitive_generator.cpp` already sets it to some value in `createPrimitiveFromControllerInput`

Comment (jonathanlew) [reply src/software/handheld_controller/controller_primitive_generator_test.cpp:83]: I'm sure you can set the x/y/angular velocities on the ControllerInput. Let me know if you have questions

File: src/software/handheld_controller/controller_primitive_generator_test.cpp [hunk 3]
```diff
@@ -28,3 +31,77 @@ TEST(ControllerPrimitiveGeneratorTest, test_create_direct_velocity)
               0.5);
     EXPECT_EQ(direct_velocity_primitive->direct_control().dribbler_speed_rpm(), 200);
 }
+
+// Tests if controller_primitive_generator returns a kick primitive if the
+// kick button is set to pressed
```
Comment (jonathanlew) [inline src/software/handheld_controller/controller_primitive_generator_test.cpp]: we usually put these comments inside the test, so after the open brace

**Review by jonathanlew — DISMISSED** [review]
lgtm other than a minor nit

**Review by akhilveeraghanta — DISMISSED** [review]
:100: 

edit: make sure to remove WIP before merging the PR

**Review by kelvinkoon — DISMISSED** [review]

**Review by jonathanlew — APPROVED** [review]
LGTM! :+1:

**Review by kelvinkoon — APPROVED** [review]
:video_game:

**Pull request merged**